src/components/IDO/IDO.tsx

```diff
@@ -0,0 +1,265 @@
+import { printBN, printBNtoBN } from '@consts/utils'
+import { OutlinedButton } from '@components/OutlinedButton/OutlinedButton'
+import watchIcon from '@static/svg/watch.svg'
+import solanaIcon from '@static/svg/solanaToken.svg'
+import dollarIcon from '@static/svg/dollarToken.svg'
+import invariantIcon from '@static/svg/invariantToken.svg'
+import invariantLogo from '@static/svg/invariantLogo.svg'
+import { SwapToken } from '@components/Swap/Swap'
+import AnimatedNumber from '@components/AnimatedNumber'
+import { Status } from '@reducers/solanaWallet'
+
+export interface IIDO {
+  tokens: SwapToken[]
+  totalDeposit: number
+  totalSolContribute: number
+  priceOfToken: number
+  valueOfInvariantTokens: number
+  walletStatus: Status
+  claimable: Boolean
+  withdrawable: Boolean
+  currencyInfo: ICurrencyData
+}
+interface ICurrencyData {
+  bitcoin: ICurrencyObject
+  tether: ICurrencyObject
+  solana: ICurrencyObject
+  ethereum: ICurrencyObject
+}
+interface ICurrencyObject {
+  usd: number
+}
+
+const IDO: React.FC<IIDO> = ({
+  tokens,
+  totalDeposit,
+  totalSolContribute,
+  priceOfToken,
+  valueOfInvariantTokens,
+  walletStatus,
+  claimable,
+  withdrawable,
+  currencyInfo
+}) => {
+  const classes = useStyles()
+
+  const [tokenFromIndex, setTokenFromIndex] = React.useState<number | null>(
+    tokens.length ? 0 : null
+  )
+  const [amountFrom, setAmountFrom] = React.useState<string>('')
+  const [totalDeposited, setTotalDeposited] = React.useState<number>(0)
+  const [totalSolContributed, setTotalSolContributed] = React.useState<number>(0)
+  const [tokenPrice, setTokenPrice] = React.useState<number>(0)
+  const [invariantForSale, setInvariantForSale] = React.useState<number>(0)
+
+  const getButtonMessage = (): string => {
+    if (walletStatus !== Status.Initialized) {
+      return 'Connect a wallet'
+    }
+
+    if (tokenFromIndex === null || tokenFromIndex === null) {
+      return 'Deposit'
+    }
+
+    if (withdrawable === true) {
+      return 'Withdraw'
+    }
+
+    if (claimable === true) {
+      return 'Claim'
+    }
+
+    if (printBNtoBN(amountFrom, tokens[tokenFromIndex].decimal).eqn(0)) {
+      return 'Insufficient trade volume'
+    }
+    if (
+      printBNtoBN(amountFrom, tokens[tokenFromIndex].decimal).gt(
+        printBNtoBN(
+          printBN(tokens[tokenFromIndex].balance, tokens[tokenFromIndex].decimal),
+          tokens[tokenFromIndex].decimal
+        )
+      )
+    ) {
+      return 'Insufficient balance'
+    }
+    return 'Deposit'
+  }
+
+  const getHeaderMessage = (): string => {
+    if (claimable === true) {
+      return 'Claim your SOL'
+    } else if (withdrawable === true) {
+      return 'Withdraw your SOL'
+    } else {
+      return 'Deposit your SOL'
+    }
+  }
+
+  const formatValue = (value: number): string => value.toFixed(0)
+
+  useEffect(() => {
+    if (tokenFromIndex !== null) {
+      setAmountFrom('0.000000')
+    }
+    setTotalDeposited(totalDeposit)
```

Comment:
these useState uses are not needed now, because you're already setting values through props